bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -73,8 +76,8 @@ public void initialize() {
         EnergyMeterConfig config = getConfigAs(EnergyMeterConfig.class);
 
         try {
-            serialNumber = config.getSerialNumber();
-            if (serialNumber == null) {
+            serialNumber = Objects.requireNonNullElse(config.getSerialNumber(), "");
```

Comment:
Why do you need to use Objects.requireNonNullElse while now config.getSerialNumber() returns a non null value ?

Reply:
`config.getSerialNumber()` is `@Nullable`

Reply:
Correct, I thought it was what you changed but in fact you kept a nullable value in the config class while you made it non nullable in the other classes.